src/main/java/com/aws/greengrass/deployment/DeploymentService.java

```diff
@@ -323,6 +323,8 @@ private void persistDeploymentQueueToConfig() {
             if (deploymentsToSave.isEmpty()) {
                 return;
             }
+            // do not persist deployments that target the nucleus itself, as doing so can cause undesired behavior.
+            deploymentsToSave.removeIf(deployment -> doesDeploymentChangeTheNucleusVersion(deployment));
```

Comment:
this can be replaced by a method reference.

Reply:
Yeah